hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app/src/main/java/org/apache/hadoop/mapreduce/v2/app/job/impl/TaskAttemptImpl.java

```diff
@@ -377,6 +377,12 @@
              TaskAttemptEventType.TA_COMMIT_PENDING,
              TaskAttemptEventType.TA_FAILMSG,
              TaskAttemptEventType.TA_FAILMSG_BY_CLIENT))
+     // Transitions from SUCCESS_FINISHING_CONTAINER state to FAILED stat
+     // When the event TA_TOO_MANY_FETCH_FAILURE received
```

Comment:
Nit: IMO this comment isn't worth it since it exactly parrots the code.  Also I'd prefer to see this transition appear before the ignored transitions in this state, as the convention has been to put ignored transitions at the end of the list of event handlers for each state.